Sources/NFCPassportReader/SecureMessaging.swift

```diff
@@ -125,7 +125,7 @@ public class SecureMessaging {
         Logger.secureMessaging.debug("\t\tSSC: \(binToHexRep(self.ssc))")
                 
         // Check for a SM error
-        if(rapdu.sw1 != 0x90 || rapdu.sw2 != 0x00) {
+        if (rapdu.sw1 != 0x90 || rapdu.sw2 != 0x00) && (rapdu.sw1 != 0x62 && rapdu.sw2 != 0x82) {
```

Comment:
Just out of curiosity: Where have you found these hex values? Is there some kind of documentation you've used?

Reply:
These hex values are from the [ISO/IEC 7816-4](https://www.freecalypso.org/pub/GSM/ISO7816/ISO_7816-4_2005.pdf) standard that is referenced by [ICAO 9303](https://www.icao.int/publications/pages/publication.aspx?docnum=9303).
`0x62 0x82` signals (page 12 of the linked PDF):
> End of file or record reached before reading Ne bytes

Generally, responses with the `SW1` being `0x62` indicate just a warning, and not an error. We get this warning, because in our case the max read size is `0xFFF`, however we get less data from the chip than that, which is why it states that there are less bytes read than the expected (=max) read size.